src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -11,300 +11,36 @@
 using System.Runtime.InteropServices;
 using System.Threading;
 using System.Threading.Tasks;
-using System.Threading.Tasks.Dataflow;
 using Nethermind.Core;
-using Nethermind.Core.Caching;
 using Nethermind.Core.Collections;
 using Nethermind.Core.Crypto;
 using Nethermind.Core.Extensions;
 using Nethermind.Logging;
 
 namespace Nethermind.Trie.Pruning
 {
-    using Nethermind.Core.Cpu;
-    using CollectionExtensions = Core.Collections.CollectionExtensions;
-
     /// <summary>
     /// Trie store helps to manage trie commits block by block.
     /// If persistence and pruning are needed they have a chance to execute their behaviour on commits.
     /// </summary>
     public class TrieStore : ITrieStore, IPruningTrieStore
     {
-        internal class DirtyNodesCache
```

Comment:
Can we move back DirtyNodesCache here for ease of review and move it later just before merge?

Reply:
ok